common/display/displayqueue.cpp

```diff
@@ -820,23 +820,18 @@ bool DisplayQueue::QueueUpdate(std::vector<HwcLayer*>& source_layers,
   }
   // Handle any 3D Composition.
   if (render_layers) {
-    if (!compositor_.BeginFrame(disable_ovelays)) {
-      ETRACE("Failed to initialize compositor.");
-      composition_passed = false;
-    }
+    compositor_.BeginFrame(disable_ovelays);
 
-    if (composition_passed) {
-      std::vector<HwcRect<int>> layers_rects;
-      for (size_t layer_index = 0; layer_index < size; layer_index++) {
-        const OverlayLayer& layer = layers.at(layer_index);
-        layers_rects.emplace_back(layer.GetDisplayFrame());
-      }
+    std::vector<HwcRect<int>> layers_rects;
+    for (size_t layer_index = 0; layer_index < size; layer_index++) {
+      const OverlayLayer& layer = layers.at(layer_index);
+      layers_rects.emplace_back(layer.GetDisplayFrame());
+    }
 
-      // Prepare for final composition.
-      if (!compositor_.Draw(current_composition_planes, layers, layers_rects)) {
-        ETRACE("Failed to prepare for the frame composition. ");
-        composition_passed = false;
-      }
+    // Prepare for final composition.
+    if (!compositor_.Draw(current_composition_planes, layers, layers_rects)) {
+      ETRACE("Failed to prepare for the frame composition. ");
+      composition_passed = false;
```

Comment:
If 'composition_passed' is no longer utilized in all 'if' conditional checks, remove the declaration, all references and assignments.

Reply:
Unfortunately it still is used for checking if these Draw calls work, so I can only get rid of checks prior to the first site that it can possibly get set to false.